code/modules/mob/mob_transformation_simple.dm

```diff
@@ -36,6 +36,14 @@
 	return M
 
 
+/mob/living/carbon/human/change_mob_type(new_type, turf/location, new_name, delete_old_mob, subspecies)
+	if(!ispath(new_type, /mob/living/carbon/human))
+		return ..()
+	var/mob/living/carbon/carbon_path = new_type
+	if(!subspecies)
+		subspecies = initial(carbon_path.species)
+	on_transformation(subspecies)
```

Comment:
Whats this doing?

Reply:
It swaps the species, rather than creating a new human mob and then transfering the mind. Check the change_mob_type just up there, it's not adapated to human to human transformation